fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -360,39 +360,18 @@ public void commitTransaction(long dbId, long transactionId, List<TabletCommitIn
                 }
             }
         }
-        
+
+        // before state transform
+        transactionState.beforeStateTransform(TransactionStatus.COMMITTED);
+        // transaction state transform
         writeLock();
         try {
-            // transaction state is modified during check if the transaction could committed
-            if (transactionState.getTransactionStatus() != TransactionStatus.PREPARE) {
-                return;
-            }
-            // 4. update transaction state version
-            transactionState.setCommitTime(System.currentTimeMillis());
-            transactionState.setTransactionStatus(TransactionStatus.COMMITTED);
-            transactionState.setErrorReplicas(errorReplicaIds);
-            for (long tableId : tableToPartition.keySet()) {
-                TableCommitInfo tableCommitInfo = new TableCommitInfo(tableId);
-                for (long partitionId : tableToPartition.get(tableId)) {
-                    OlapTable table = (OlapTable) db.getTable(tableId);
-                    Partition partition = table.getPartition(partitionId);
-                    PartitionCommitInfo partitionCommitInfo = new PartitionCommitInfo(partitionId,
-                                                                                      partition.getNextVersion(),
-                                                                                      partition.getNextVersionHash());
-                    tableCommitInfo.addPartitionCommitInfo(partitionCommitInfo);
-                }
-                transactionState.putIdToTableCommitInfo(tableId, tableCommitInfo);
-            }
-            // 5. persistent transactionState
-            unprotectUpsertTransactionState(transactionState);
-            
-            // add publish version tasks. set task to null as a placeholder.
-            // tasks will be created when publishing version.
-            for (long backendId : totalInvolvedBackends) {
-                transactionState.addPublishVersionTask(backendId, null);
-            }
+            unprotectedCommitTransaction(transactionState, errorReplicaIds, tableToPartition, totalInvolvedBackends,
+                                         db);
         } finally {
             writeUnlock();
+            // after state transform
+            transactionState.afterStateTransform(TransactionStatus.COMMITTED);
```

Comment:
If fail to commit, call `afterStateTransform` with COMMITTED is OK?

Reply:
I will add a boolean variable to judge correctly commit txn